lib/THCUNN/generic/SpatialAveragePooling.cu

```diff
@@ -27,11 +27,6 @@ static inline void THNN_(SpatialAveragePooling_shapeCheck)(
 
   THCUNN_argCheck(state, ndim == 3 || ndim == 4, 2, input,
                   "3D or 4D input tensor expected but got: %s");
-
```

Comment:
Let's take the most general case involving padding and dilation.  I believe the intention of the check is to make sure that the effective input size (inputsize + 2 * padding) is >= the effective kernel size (dilation * (k-1) + 1).  If that's the case, the removed check (inputsize >= k - pad) is wrong, because e.g. inputsize=1, pad=1, k=3, dilation=1 is a valid input but fails the check (and runs successfully if the check is removed).

Then, the "corrected" check above is equivalent to the existing check below that the output dimension is >= 1.